### ISSUE TITLE
lambda.Function Error: Cannot configure 'securityGroup' or 'allowAllOutbound' without configuring a VPC

### ISSUE THREAD

**Issue by shishkin** [NickBurkard/aws-cdk-scala#51]
Hi! Thanks for creating this useful project! Unfortunately, I'm already facing an issue trying a simple example:

```scala
import io.burkard.cdk._
import io.burkard.cdk.services.lambda._

object ExampleApp extends CdkApp {
  CdkStack(id = Some("ScalaStack")) { implicit stackCtx =>

    Function(
      internalResourceId = "ScalaHandler",
      code = AssetCode(
        path = "lambda/target/universal/lambda-0.1.0-SNAPSHOT.zip"
      ),
      handler = "lambda.handler",
      runtime = Runtime("nodejs14.x", RuntimeFamily.Nodejs)
    )
  }
}
```

This throws an error when doing `cdk synth`: Error: Cannot configure 'securityGroup' or 'allowAllOutbound' without configuring a VPC.

```
[info] running ExampleApp
Exception in thread "sbt-bg-threads-1" java.lang.ExceptionInInitializerError
	at ExampleApp.main(ExampleApp.scala)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at sbt.Run.invokeMain(Run.scala:143)
	at sbt.Run.execute$1(Run.scala:93)
	at sbt.Run.$anonfun$runWithLoader$5(Run.scala:120)
	at sbt.Run$.executeSuccess(Run.scala:186)
	at sbt.Run.runWithLoader(Run.scala:120)
	at sbt.Defaults$.$anonfun$bgRunMainTask$7(Defaults.scala:1951)
	at sbt.Defaults$.$anonfun$termWrapper$2(Defaults.scala:1922)
	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
	at scala.util.Try$.apply(Try.scala:213)
	at sbt.internal.BackgroundThreadPool$BackgroundRunnable.run(DefaultBackgroundJobService.scala:369)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: software.amazon.jsii.JsiiException: Cannot configure 'securityGroup' or 'allowAllOutbound' without configuring a VPC
Error: Cannot configure 'securityGroup' or 'allowAllOutbound' without configuring a VPC
    at Function.configureVpc (/private/var/folders/3v/xv1cb7c56n70zy74nqfdv9w00000gn/T/jsii-kernel-6zPQph/node_modules/aws-cdk-lib/aws-lambda/lib/function.js:1:12848)
    at new Function (/private/var/folders/3v/xv1cb7c56n70zy74nqfdv9w00000gn/T/jsii-kernel-6zPQph/node_modules/aws-cdk-lib/aws-lambda/lib/function.js:1:4479)
    at /private/var/folders/3v/xv1cb7c56n70zy74nqfdv9w00000gn/T/jsii-java-runtime12803932902832032970/lib/program.js:8432:58
    at Kernel._wrapSandboxCode (/private/var/folders/3v/xv1cb7c56n70zy74nqfdv9w00000gn/T/jsii-java-runtime12803932902832032970/lib/program.js:8860:24)
    at Kernel._create (/private/var/folders/3v/xv1cb7c56n70zy74nqfdv9w00000gn/T/jsii-java-runtime12803932902832032970/lib/program.js:8432:34)
    at Kernel.create (/private/var/folders/3v/xv1cb7c56n70zy74nqfdv9w00000gn/T/jsii-java-runtime12803932902832032970/lib/program.js:8173:29)
    at KernelHost.processRequest (/private/var/folders/3v/xv1cb7c56n70zy74nqfdv9w00000gn/T/jsii-java-runtime12803932902832032970/lib/program.js:9757:36)
    at KernelHost.run (/private/var/folders/3v/xv1cb7c56n70zy74nqfdv9w00000gn/T/jsii-java-runtime12803932902832032970/lib/program.js:9720:22)
    at Immediate._onImmediate (/private/var/folders/3v/xv1cb7c56n70zy74nqfdv9w00000gn/T/jsii-java-runtime12803932902832032970/lib/program.js:9721:46)
    at processImmediate (node:internal/timers:464:21)
	at software.amazon.jsii.JsiiRuntime.processErrorResponse(JsiiRuntime.java:124)
	at software.amazon.jsii.JsiiRuntime.requestResponse(JsiiRuntime.java:95)
	at software.amazon.jsii.JsiiClient.createObject(JsiiClient.java:89)
	at software.amazon.jsii.JsiiEngine.createNewObject(JsiiEngine.java:603)
	at software.amazon.awscdk.services.lambda.Function.<init>(Function.java:52)
	at software.amazon.awscdk.services.lambda.Function$Builder.build(Function.java:1014)
	at io.burkard.cdk.services.lambda.Function$.apply(Function.scala:85)
	at ExampleApp$.$anonfun$new$1(ExampleApp.scala:11)
	at io.burkard.cdk.CdkStack$$anon$1.$anonfun$new$1(CdkStack.scala:37)
	at io.burkard.cdk.package$ValueDiscard$.apply(package.scala:19)
	at io.burkard.cdk.CdkStack$$anon$1.<init>(CdkStack.scala:37)
	at io.burkard.cdk.CdkStack$.apply(CdkStack.scala:37)
	at ExampleApp$.<clinit>(ExampleApp.scala:9)
	... 18 more
```

While the same example with Java CDK runs:

```java
import software.amazon.awscdk.App;
import software.amazon.awscdk.Stack;
import software.amazon.awscdk.StackProps;
import software.amazon.awscdk.services.lambda.Code;
import software.amazon.awscdk.services.lambda.Function;
import software.amazon.awscdk.services.lambda.FunctionProps;
import software.amazon.awscdk.services.lambda.Runtime;
import software.constructs.Construct;

public class JavaStack extends Stack {

  public JavaStack(final Construct parent, final String id, final StackProps props) {
    super(parent, id, props);

    new Function(
      this,
      "JavaHandler",
      FunctionProps
        .builder()
        .runtime(Runtime.NODEJS_14_X)
        .code(Code.fromAsset("lambda/target/universal/lambda-0.1.0-SNAPSHOT.zip"))
        .handler("lambda.handler")
        .build());
  }

  public static void main(final String[] args) {
    App app = new App();
    new JavaStack(app, "LambdaPackagingStack", null);
    app.synth();
  }
}
```

Looking at how `securityGroups` and `allowAllOutbound` are configured in the library:

```scala
      .allowAllOutbound(allowAllOutbound.map(Boolean.box).getOrElse(java.lang.Boolean.FALSE))
      .securityGroups(securityGroups.map(_.asJava).orNull)
```

I think there might be an issue in handling empty values in the builder. Unfortunately I couldn't get through the Java CDK proxy source code to debug it further.

**Comment by shishkin** [comment]
Maybe `allowAllOutbound` should be `null` instead of `false` when not specified?

**Comment by NickBurkard** [comment]
Thanks for finding this @shishkin, I'll dig into the source code to see what the CDK is setting default values for Java booleans as and try to reflect that in the DSL with a PR.

**Comment by NickBurkard** [comment]
Should be resolved in [0.9.1](https://github.com/NickBurkard/aws-cdk-scala/releases/tag/v0.9.1), let me know if the issue persists! 😃

**Comment by shishkin** [comment]
Thanks for the very quick response! Yes, it works perfectly now 🤩
